cmd/maya-exporter/app/collector/jivacollector.go

```diff
@@ -183,11 +187,26 @@ func (e *VolumeExporter) Describe(ch chan<- *prometheus.Desc) {
 // concurrent readers.
 
 // Collect collects all the registered stats metrics from the OpenEBS volumes.
+// It tries to reconnect with the volume if there is any error via a goroutine.
 func (e *VolumeExporter) Collect(ch chan<- prometheus.Metric) {
-	if err := e.collect(); err != nil {
-		return
+	// verify if controller url is not empty
+	if len(e.VolumeControllerURL) != 0 {
+		// collect the metrics from jiva controller and send it via channels
+		if err := e.collect(); err != nil {
+			glog.Error("Error in collecting metrics, found error:", err)
+			return
+		}
+	}
+	// verify if net.Conn is not nil and proceed to collect metrics from
+	// socket.
+	if e.Conn != nil {
+		if err := collect(e); err != nil {
```

Comment:
It is little confusing to have e.collect() & collect(e)
Will it be good to have these as methods to VolumeExporter struct?

Reply:
"e.collect()" is used for the jiva and "collect()" is used for the cstor, to reuse the code i didn't made changes otherwise i will have to make a two different structures (i.e, instead of VolumeExporter, i will have to use CstorExporter and JivaExporter structs)
as both the function does the same thing but with different business logic, i think the name itself speaks about its purpose. What do you say ? should we go with the above approach (two different structs or with the same name but with different signature or with different names ?)